gui/application.cc

```diff
@@ -42,6 +42,10 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
     QSurfaceFormat fmt;
     fmt.setSamples(10);
     fmt.setProfile(QSurfaceFormat::CoreProfile);
+    // macOS is very picky about this version matching
+    // the version of openGL  used in ImGuiRenderer
+    fmt.setMajorVersion(3);
+    fmt.setMinorVersion(2);
```

Comment:
Removing the setMajorVersion from `https://github.com/YosysHQ/nextpnr/blob/master/gui/fpgaviewwidget.cc#L62-L65` while this patch is applied doesn't break anything on the first glance.

According to [Qt5 docs](http://doc.qt.io/qt-5/qsurfaceformat.html#setDefaultFormat), it's better to set it globally than on per-widget basis.

>It can always be overridden on a per-instance basis by using the class in question's own setFormat() function. However, it is often more convenient to set the format for all windows once at the start of the application. It also guarantees proper behavior in cases where shared contexts are required, because settings the format via this function guarantees that all contexts and surfaces, even the ones created internally by Qt, will use the same format.

Reply:
I've removed the setting of this in fpgaviewwidget.cc - there was some error checking which is now gone. Possibly we should add this elsewhere but I think this is beyond the scope of this patch.

Reply:
Please copy the checking which you removed from `fpgaviewwidget.cc` here, otherwise LGTM.